packages/coingecko/src/query/routes/coinHistory.ts

```diff
@@ -0,0 +1,113 @@
+import { JSON } from "@web3api/assemblyscript-json";
+import { COINGECKO_API_URL } from "../config";
+import {
+  getIntegerProperty,
+  getNullableIntegerProperty,
+  getNullableStringProperty,
+  getStringProperty,
+  normalizeCurrencyMarketCapPairObject,
+  normalizeCurrencyPricePairObject,
+  normalizeCurrencyVolumePairObject,
+  normalizeLocalizationPairObject,
+} from "../utils";
+import {
+  Input_coinHistory,
+  CoinHistory,
+  HTTP_UrlParam,
+  HTTP_Query,
+  HTTP_ResponseType,
+} from "../w3";
+
+export function coinHistory(input: Input_coinHistory): CoinHistory {
+  const url = COINGECKO_API_URL + "/coins/" + input.id + "/history";
+  const urlParams: Array<HTTP_UrlParam> = [{ key: "date", value: input.date }];
```

Comment:
Can you add a validator here to check whether date is in valid format? Ref: https://www.assemblyscript.org/stdlib/date.html#constructor

Reply:
Thanks for the feedback, I've added validation for date there.